Sources/NIOCore/SocketAddresses.swift

```diff
@@ -394,6 +394,47 @@ public enum SocketAddress: CustomStringConvertible, NIOSendable {
         }
     }
 
+    /// Creates a new `SocketAddress` corresponding to the netmask for a subnet prefix.
+    ///
+    /// As an example, consider the subnet "127.0.0.1/8". The "subnet prefix" is "8", and the corresponding netmask is "255.0.0.0".
+    /// This initializer will produce a `SocketAddress` that contains "255.0.0.0".
+    ///
+    /// - parameters:
+    ///     - prefix: The prefix of the subnet.
+    /// - returns: A `SocketAddress` containing the associated netmask.
+    internal init(ipv4MaskForPrefix prefix: Int) {
```

Comment:
Do we want to `precondition`/`assert` that `prefix` is valid?

Reply:
Yeah good question. I wasn't honestly sure. Given that these are `internal` I considered whether we should just allow them to produce gibberish, but maybe we should precondition instead.

Reply:
Updated.